Automattic/wp-calypso
# Guided onboarding: add intent based plans

Related to 7019-gh-Automattic/dotcom-forge

## Proposed Changes

* Add mapping to show relevant plans based on user segment.

It should follow this table

![image](https://github.com/Automattic/wp-calypso/assets/17054134/f412b04e-dc9b-4b04-bbaa-d0560512aaca)

P2 Segment to Plan mapping: pdDR7T-1xi-p2

## Why are these changes being made?
See pdDR7T-1xi-p2

## Testing Instructions

1. Go to /start/guided
2. The answers to the survey should follow the following tables and the plans should match


Intent (Q1) | Goals (Q2) | Segment | Plans or action
-- | -- | -- | --
Migrate or import | n/a | Migration | Redirect to import
Create for clients | n/a | Developer / Agency | Creator, Entrepreneur, Enterprise
Create for self | Unknown (question skipped) | Unknown | All plans
Create for self | includes Get a website built quickly | DIFM | Redircet to DIFM
Create for self | Sell something and not Get a website built quickly | Merchant | Creator, Entrepreneur, Enterprise
Create for self | not Get a website built quickly and not Sell something | Consumer / Business | Starter, Explorer, Creator
Create for self | Publish a blog | Blogger | Free, Starter, Explorer, Creator
Create for self | Educational/nonprofit | Nonprofit | Free, Starter, Explorer, Creator
